### PR TITLE
Git_Basics.rst: Add "Run coala" section

### PR DESCRIPTION
**Pull request by corona10**
Tell people to run coala before pushing.

Closes https://github.com/coala/coala/issues/3874

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by twinone** [comment]
ack f6cb5a65551d8a7a0f2395fadb4e06b7332b070e

**Comment by NiklasMM** [comment]
unack f6cb5a6

File: docs/Developers/Git_Basics.rst [hunk 1]
```diff
@@ -163,6 +163,18 @@ Now that your message is written, you will have to save the file. Press escape
 to exit insert mode, and save the file (in Vim that is being done by pressing
 shift + Z twice).
 
+Run Coala
+------------------
+
+Now you can check if your commit messages and the code formattings are
+appropriate by using coala.
+If something goes wrong, coala will let you know. If this fails, CI
```
Comment (NiklasMM) [inline docs/Developers/Git_Basics.rst]: I don't think we can assume everyone knows what CI stands for. Maybe change it to `continuous integration (CI)`?

Comment (corona10) [inline docs/Developers/Git_Basics.rst]: @NiklasMM Coool!

**Comment by corona10** [comment]
@NiklasMM PTAL

File: docs/Developers/Git_Basics.rst [hunk 2]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
 to exit insert mode, and save the file (in Vim that is being done by pressing
 shift + Z twice).
 
+Run Coala
+------------------
+
+Now you can check if your commit messages and the code formattings are
```
Comment (Udayan12167) [inline docs/Developers/Git_Basics.rst]: I would reword to " you can check if .... conform with community guidelines"

File: docs/Developers/Git_Basics.rst [hunk 3]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
...
+appropriate by using coala.
+If something goes wrong, coala will let you know. If this fails,
+continuous integration (CI) also fails. It means that you can not proceed
```
Comment (Udayan12167) [inline docs/Developers/Git_Basics.rst]: Rather "you cannot proceed with merging your fix/pull request"

**Comment by corona10** [comment]
@Udayan12167 Thanks for the reviews. I updated it.

File: docs/Developers/Git_Basics.rst (hunk 2 above)
Comment (Udayan12167) [inline docs/Developers/Git_Basics.rst]: Remove "are"

**Comment by corona10** [comment]
@Udayan12167 PTAL 😄

**Comment by supergr35** [comment]
Please capitalize the `c` in `closes` in your commit message.

**Comment by corona10** [comment]
@supergr35 Thanks! Done.

**Comment by supergr35** [comment]
Actually, your commit message could probably better describe your change. Something more like: `Change Git_Basics.rst to tell people to run coala over their code before pushing.` 
With the right new lines, of course.

**Comment by corona10** [comment]
@supergr35 You are definitely right. I changed a title also. Thanks.

**Comment by supergr35** [comment]
ack b59520b

File: docs/Developers/Git_Basics.rst [hunk 4]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
 to exit insert mode, and save the file (in Vim that is being done by pressing
 shift + Z twice).
 
+Run Coala
+------------------
+
+Now you can check if your commit messages and the code formattings
+conform with community guidelines.
+If something goes wrong, coala will let you know. If this fails,
+continuous integration (CI) also fails. It means that you cannot proceed
```
Comment (yash-nisar) [inline docs/Developers/Git_Basics.rst]: you cannot proceed -> we cannot proceed ?

Comment (yash-nisar) [inline docs/Developers/Git_Basics.rst]: Can we write `The continuous integration (CI) will fail if coala reports errors.` instead of `If this fails, continuous integration (CI) also fails`

File: docs/Developers/Git_Basics.rst [hunk 5]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
 to exit insert mode, and save the file (in Vim that is being done by pressing
 shift + Z twice).
 
+Run Coala
+------------------
+
+Now you can check if your commit messages and the code formattings
```
Comment (yash-nisar) [inline docs/Developers/Git_Basics.rst]: Redundant `the` before `code formattings` -> `Now you can check if your commit messages and code formattings`

File: docs/Developers/Git_Basics.rst [hunk 6]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
...
+conform with community guidelines.
```
Comment (yash-nisar) [inline docs/Developers/Git_Basics.rst]: Add `the` between `with` and `community` -> `conform with the community guidelines`

**Comment by corona10** [comment]
@yash-nisar Thanks for the comment. PTAL

File: docs/Developers/Git_Basics.rst [hunk 7]
```diff
@@ -163,6 +163,19 @@ Now that your message is written, you will have to save the file. Press escape
 to exit insert mode, and save the file (in Vim that is being done by pressing
 shift + Z twice).
 
+Run Coala
```
Comment (NiklasMM) [inline docs/Developers/Git_Basics.rst]: coala is always written with a lower case `c` ;-)

**Comment by NiklasMM** [comment]
Your shortlog: "Git_Basics.rst: Add explains for using coala" sounds a bit weird. How about 'Git_Basics.rst: Add "Run coala" section'

**Comment by corona10** [comment]
@NiklasMM PTAL

**Comment by NiklasMM** [comment]
ack 23211b5

**Comment by NiklasMM** [comment]
Looks good! Please rebase and we can go ahead with the merge

**Comment by corona10** [comment]
@NiklasMM
Thanks! Rebase done!

**Comment by NiklasMM** [comment]
reack f40e524

**Comment by NiklasMM** [comment]
@rultor merge
